### PR TITLE
Chaplain's Holyparasite

### PR DESCRIPTION
**Pull request by ynot01**
### Intent of your Pull Request

fun

It heals at half the rate of the support holoparasite and deals about half the damage of a null rod, in exchange for your null rod. You are supposed to keep the deck of holy tarot cards as they retain the antimagic property of a null rod. The holyparasite also has the holy trait, deflecting magic.

custom sprite courteousy of @sam0oneau 

96EE56B

balance suggestions open

#### Changelog

:cl:  
rscadd: Added a holyparasite to the null rod reskins
/:cl:

### CODE REVIEW

**Comment by Szyszkrzyneczka** [comment]
KOREGA... REQUIEM DA

**Comment by drderp3635** [comment]
Stupidly strong. 

👎

**Comment by ynot01** [comment]
> Stupidly strong.
> 
> 👎

elaborate

also you seem to have missed the
> balance suggestions open

**Comment by drderp3635** [comment]
Round start holoparasite is strong because its a fucking roundstart nonantag holoparasite even if its half healing and null rod damage. Chaplain still gets antimagic with the deck, and the lower damage doesnt even matter because the chaplain can still be hitting the same target with a crowbar or some easily obtainable shit to deal more damage than any other null rod.
The healing is just the cherry on top for how strong this is because you are a one man medbay as someone who is supposed to be a rp role.

If you want to balance this take antimagic off the deck, and completely remove the healing.

**Comment by ynot01** [comment]
your bible already heals, so it's not like chaplain is new to healing, antimagic for the holypara is just so that chaplain doesnt get fucked by wizard spells where he'd normally be resistant

originally i didnt have it dealing damage at all, but folks in #coder wanted a balance to sacrificing your powercreep weapon, so i added some damage

once again, if you think any of the following values should be changed to better balance:
2.5 heal of brute/burn/toxin/oxy when in healer mode
100% damage transfer when in healer mode
10 damage to mobs and objects when in combat mode
90% damage transfer when in combat mode

please let me know

**Comment by drderp3635** [comment]
Bible healing is garbage, and inflicts brain damage

**Comment by ynot01** [comment]
> Bible healing is garbage, and inflicts brain damage

not if you're wearing a helmet

**Comment by jcatjmeow** [comment]
I say remove the healing, I agree it's op. Even if you think the rate is fine, remember that the holoparasite can drag a chaplain's crit body out of danger and then heal it out of crit.

**Comment by Codeatmos** [comment]
I don't think its (that) strong,clock it down to 7 damage,remove anti-magic,BUT OH WAIT,ITS NOT OP
Holoparas are slower then humans
It heals 2.5 per punch,what the fuck are you gonna outheal?
If you die,you get DUSTED
It`s a second hitbox
It will not save your ass,its slower then most antags.
You dont have a null-rod,which means less deconversion of cult items.

**Comment by Ratvar** [comment]
I propose making host of holyparasite to have antimagic instead of deck. This way you cannot give away tarot to HoS, which you are encouraged to do because tarot is useless for you now.

I encourage you to give chaplain PACIFISM braintrauma and to give holypara full nullrod damage. This way you trade off being able to attack staff like human in exchange for ability to reposition, block non-directly damaging projectiles and attack while critted. And also solve a problem of Holypara dealing 10 damage and you dealing 18 Spear damage at one. Will also make it closer to Jojo.        


   **If host's already pacifist, make holopara pacifist as well: no gaming the system for trait points.**
Make it touch inflict nullrod effects on cultists and cult structures: cleansing items, deleting runes, etc.
How to make it delete runes:
add **if(istype(M, /mob/living/simple_animal/hostile/guardian/chaplain)) 
            to_chat(M, "<span class='warning'>You purge the rune!</span>")
            qdel(src)**
to  **code/modules/antagonists/cult/runes.dm**
and holypara will delete cult runes on hit
right after **/obj/effect/rune/attack_animal(mob/living/simple_animal/M)
    if(istype(M, /mob/living/simple_animal/shade) || istype(M, /mob/living/simple_animal/hostile/construct))**
and before **else
            to_chat(M, "<span class='warning'>You are unable to invoke the rune!</span>")**

**Comment by jcatjmeow** [comment]
"what the fuck are you gonna outheal?"
Oxyloss and bleeding damage from being in crit

**Comment by Ratvar** [comment]
Jcat, we even talked on discord about removing ability to heal yourself/chaplain, stop pretending
Adding to that, do remove ability to self-heal as holypara: use pacifist code bits or something, plz.

**Comment by MonsWTF** [comment]
Ahh, something else Shitcurity can arrest the chaplain over!  /s

I do think that the healing could use the trade-off of the bible's healing of causing some amount of brain damage, even if the bible only does it in the absence of a helmet (see: majority of the time). 

0.5 to 1 brain damage per healing touch?

Edit: If Ynot had connections to someone that can sprite, maybe an appropriately-holy-themed version of Holopara to distinguish it from standard HP's. Unless that's part of its downsides, I have a feeling there will be a significant # of people confusing it for the tator tot version and murderboning the Chappie the first time it's seen.

**Comment by yacabo** [comment]
Give it a new sprite so that sec can tell the difference between syndicate holo and chap holo

**Comment by ynot01** [comment]
should be noted that chaplain's holypara is not skinned as tator holopara, but is skinned as the guardian spirit that wizard spawns and the same one you can get from lavaland

**Comment by fluffe9911** [comment]
neat

**Comment by ShadowDeath6** [comment]
no healing, chaplain doesn't need a way to heal itself especially when it brings an unstunnable body that can drag people around

Just cut out healing and make it do more damage (like 12 or something)

**Comment by Ratvar** [comment]
I would like to remind people, again, that ynot didn't prevent holypara from healing itself or chaplain only because he doesn't know how yet.

**Comment by ghost** [comment]
imagine getting a free holoparasite at roundstart

**Review by AsV9 — CHANGES_REQUESTED** [review]
I think the healing aspect of this is just far too strong in it's current state, and I'd like to see that removed or changed somehow.

**Comment by Ratvar** [comment]
I would like to remind people, again, that ynot didn't prevent holypara from healing itself or chaplain only because he doesn't know how **yet**.

**Comment by ShadowDeath6** [comment]
if(target == master)
return

**Comment by Ratvar** [comment]
It would be much more helpful if you actually would show where to put that thing instead of showing that you know how and leaving, really. No, ynot's not a great byond coder and you know it, message alone doesn't help at all.

File: code/modules/mob/living/simple_animal/guardian/types/chaplain.dm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/mob/living/simple_animal/hostile/guardian/chaplain/Initialize()
+	. = ..()
+	var/datum/atom_hud/medsensor = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+	medsensor.add_hud_to(src)
+	ADD_TRAIT(src, TRAIT_HOLY, SPECIES_TRAIT)
+
+/mob/living/simple_animal/hostile/guardian/chaplain/AttackingTarget()
+	. = ..()
+	if(is_deployed() && toggle && iscarbon(target))
+		var/mob/living/carbon/C = target
```
Comment (ShadowDeath6) [inline code/modules/mob/living/simple_animal/guardian/types/chaplain.dm:28]: if(C == summoner)
return

File: code/modules/mob/living/simple_animal/guardian/types/chaplain.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		C.adjustBruteLoss(-2.5)
+		C.adjustFireLoss(-2.5)
+		C.adjustOxyLoss(-2.5)
+		C.adjustToxLoss(-2.5)
+		var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(C))
+		if(namedatum)
+			H.color = namedatum.colour
+		if(C == summoner)
```
Comment (ShadowDeath6) [inline code/modules/mob/living/simple_animal/guardian/types/chaplain.dm:29]: get rid of this stuff

**Comment by ynot01** [comment]
will put these changes in sometime, been busy with college

**Review by AsV9 — COMMENTED** [review]
I'm still skeptical, but I suppose we can give this a try and tweak it later as needed.

**Pull request merged**